arelle/plugin/validate/ESEF/ESEF_Current/DTS.py

```diff
@@ -204,9 +204,13 @@ def lc3wordAdjust(word: str) -> str:
                 _("Extension taxonomy MUST NOT define typed dimensions: %(concepts)s."),
                 modelObject=typedDimsInExtTxmy, concepts=", ".join(str(c.qname) for c in typedDimsInExtTxmy))
         if domainMembersWrongType:
+            xbrlReference322 = "https://www.xbrl.org/dtr/type/2020-01-21/types.xsd"
+            if getDisclosureSystemYear(val.modelXbrl) < 2023:
+                xbrlReference322 = "http://www.xbrl.org/dtr/type/nonNumeric-2009-12-16.xsd"
             val.modelXbrl.error("ESEF.3.2.2.domainMemberWrongDataType",
-                _("Domain members MUST have domainItemType data type as defined in \"http://www.xbrl.org/dtr/type/nonNumeric-2009-12-16.xsd\": concept %(concepts)s."),
-                modelObject=domainMembersWrongType, concepts=", ".join(str(c.qname) for c in domainMembersWrongType))
+                _("Domain members MUST have domainItemType data type as defined in \"%(xbrlReference)s\": concept %(concepts)s."),
+                modelObject=domainMembersWrongType, xbrlReference=xbrlReference322,
+                concepts=", ".join(str(c.qname) for c in domainMembersWrongType))
```

Comment:
Older than 2023 should be [either of these schemas](https://github.com/Arelle/Arelle/blob/0975615971ef93de11d19b59bc5cf8750bb1b9d1/arelle/plugin/validate/ESEF/Const.py#L83-L91)?

I think `esefDomainItemTypes` should be moved up [out of its current location](https://github.com/acsone/Arelle/blob/f781a482e0e415405c222d413927b6ded13dd7b0/arelle/plugin/validate/ESEF/ESEF_Current/DTS.py#L112-L112) to where it can be accessed from here:
```suggestion
            val.modelXbrl.error("ESEF.3.2.2.domainMemberWrongDataType",
                _("Domain members MUST have domainItemType data type as defined in \%(xbrlReferences)\: concept %(concepts)s."),
                modelObject=domainMembersWrongType,
                xbrlReferences=" or ".join(sorted(esefDomainItemTypes)),
                concepts=", ".join(str(c.qname) for c in domainMembersWrongType))
```

Reply:
Hello,

The error message is taken straight from the ESEF Reporting Manual.

In the 2022 version (and prior) one can read:
> Domain members MUST have domainItemType data type as defined in
>
> http://www.xbrl.org/dtr/type/nonNumeric-2009-12-16.xsd

In the ESEF Reporting manual, version 2023 the meassage has been altered to:
> Domain members MUST have domainItemType data type as defined in
>
> https://www.xbrl.org/dtr/type/2020-01-21/types.xsd

These are not the values specified in `esefDomainItemTypes`.

Best regards,

Greg